tools/sigma/backends/ala.py

```diff
@@ -257,7 +265,7 @@ def generateMapItemNode(self, node):
                 elif callable(mapping):
                     return self.generateSubexpressionNode(
                             self.generateANDNode(
-                                [cond for cond in mapping(key, self.cleanValue(value))]
+                                [cond for cond in mapping(key, self.cleanValue(str(value)))]
```

Comment:
Let's not wrap these in [`str()`](https://docs.python.org/3/library/functions.html#func-str) here.  Create a new method on the class called `cleanValue` and call the parent cleanValue(str(val)) using [`super()`](https://docs.python.org/3/library/functions.html#super).

An example of this can be seen [here](https://github.com/Neo23x0/sigma/blob/11c216629b98ed3da323532d27a13934d4e6a8fa/tools/sigma/backends/misc.py#L32-L33).

This new class method may look similar to the following

```python
    def cleanValue(self, val):
        return super().cleanValue(str(val))
```

```suggestion
                                [cond for cond in mapping(key, self.cleanValue(value))]
```